code/modules/crafting/craft.dm

```diff
@@ -129,26 +130,49 @@
 			return FALSE
 	return TRUE
 
+/datum/personal_crafting/proc/check_pathtools(mob/user, datum/crafting_recipe/R, list/contents)
+	if(!R.pathtools.len) //does not run if no tools are needed
+		return TRUE
+	var/list/other_possible_tools = list()
+	for(var/obj/item/I in user.contents) // searchs the inventory of the mob
+		if(istype(I, /obj/item/storage))
+			for(var/obj/item/SI in I.contents)
+				other_possible_tools += SI.type	// filters type paths
+		other_possible_tools += I.type
+	
+	other_possible_tools |= contents["other"] // This check for type paths
+	main_loop: // checks if all tools found are usable with the recipe
+		for(var/A in R.pathtools)
+			for(var/I in other_possible_tools)
+				if(ispath(I,A))
```

Comment:
Need to include an else to this due to contents["other"] being a list of items.
So something like:
```suggestion
				if(ispath(I,A))
					continue main_loop
				else if(isitem(I) && ispath(I.path, A)
					continue main_loop
```
Or put both checks in one line